packages/sync-actions/test/types-sync-base.spec.js

```diff
@@ -0,0 +1,99 @@
+import clone from '../src/utils/clone'
+import createSyncTypes, { actionGroups } from '../src/types'
+import { baseActionsList } from '../src/types-actions'
+
+describe('Exports', () => {
+  test('action group list', () => {
+    expect(actionGroups).toEqual(['base', 'attributes'])
+  })
+
+  test('correctly define base actions list', () => {
+    expect(baseActionsList).toEqual([
+      { action: 'changeKey', key: 'key' },
+      { action: 'changeName', key: 'name' },
+      { action: 'setDescription', key: 'description' },
+    ])
+  })
+})
+
+describe('Actions', () => {
+  let typesSync
+  let updateActions
+  let before
+  let now
+  beforeEach(() => {
+    typesSync = createSyncTypes()
+  })
+  describe('mutation', () => {
+    test('should ensure given objects are not mutated', () => {
+      before = {
+        name: 'Orwell',
+        key: 'War-is-Peace',
+      }
+      now = {
+        name: 'Orwell',
+        key: 'Freedom-is-slavery',
+      }
+      typesSync.buildActions(now, before)
+      expect(before).toEqual(clone(before))
+      expect(now).toEqual(clone(now))
+    })
+  })
+  describe('with name change', () => {
+    beforeEach(() => {
+      before = {
+        name: 'Orwell',
+      }
+      now = {
+        name: 'Ignorance-is-Strength',
+      }
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `changeName` update-action', () => {
+      expect(updateActions).toEqual([
+        {
+          action: 'changeName',
+          name: 'Ignorance-is-Strength',
+        },
+      ])
+    })
+  })
+  describe('with key change', () => {
+    beforeEach(() => {
+      before = {
+        key: 'orwell-key',
+      }
+      now = {
+        key: 'huxley-key',
+      }
+      updateActions = typesSync.buildActions(now, before)
+    })
+    test('should return `changeKey` update-action', () => {
```

Comment:
I would like to see test `null -> ''` or `'' -> null`, given shouldOmitEmptyString